modules/repo_sync/tests/src/Functional/ImportTest.php

```diff
@@ -0,0 +1,152 @@
+ */
+class ImportTest extends RepoSyncTestBase {
+
+  /**
+   * Address of the background HTTP server.
+   *
+   * @var string
+   */
+  public $serverAddr = '0.0.0.0:9999';
+
+  /**
+   * Import CRUD test.
+   */
+  public function testCRUD() {
+    $this->drupalLogin($this->rootUser);
+    $label = $this->getRandomGenerator()->name();
+    $repo_url = 'https://github.com/tamasd/git-test.git';
+    $base_path = '/content/import/test';
+    $pattern = 'docs/';
+    $reference = 'refs/heads/master';
+
+    // Create.
+    $this->drupalPostForm('/admin/devportal/repo-sync/create', [
+      'label' => $label,
+      'repository_url' => $repo_url,
+      'pattern' => $pattern,
+      'reference' => $reference,
+      'base_path' => $base_path,
+    ], 'Save');
+    $uuid = $this->getUuid();
+    $this->assertNotEmpty($uuid);
+    $this->cleanUp[] = $uuid;
+
+    // View.
+    $this->assertSession()->pageTextContains($label);
+    $this->assertSession()->pageTextContains($repo_url);
+    $this->assertSession()->pageTextContains($base_path);
+    $this->assertSession()->pageTextContains($pattern);
+    $this->assertSession()->pageTextContains($reference);
+
+    // List.
+    $this->drupalGet('/admin/devportal/repo-sync/content');
+    $this->assertSession()->pageTextContains($label);
+    $this->clickLink($label);
+
+    // Update.
+    $this->clickLink('Edit');
+    $new_label = $this->getRandomGenerator()->name();
+    $this->drupalPostForm(NULL, [
+      'label' => $new_label,
+    ], 'Save');
+    $this->assertSession()->pageTextNotContains($label);
+    $this->assertSession()->pageTextContains($new_label);
+    $this->assertSession()->pageTextContains($repo_url);
+    $this->assertSession()->pageTextContains($base_path);
+    $this->assertSession()->pageTextContains($pattern);
+    $this->assertSession()->pageTextContains($reference);
+
+    // Delete.
+    $this->clickLink('Delete');
+    $this->drupalPostForm(NULL, [], 'Confirm');
+
+    // List.
+    $this->assertSession()->pageTextNotContains($label);
+    $this->assertSession()->pageTextNotContains($new_label);
+  }
+
+  /**
+   * Tests the import process.
+   */
+  public function testImport() {
+    $this->drupalLogin($this->rootUser);
+    $label = $this->getRandomGenerator()->name();
+    $repo_url = 'https://github.com/tamasd/git-test.git';
+    $base_path = '/content/import/test';
+    $pattern = 'docs/';
+    $reference = 'refs/heads/master';
+
+    // Create.
+    $this->drupalPostForm('/admin/devportal/repo-sync/create', [
+      'label' => $label,
+      'repository_url' => $repo_url,
+      'pattern' => $pattern,
+      'reference' => $reference,
+      'base_path' => $base_path,
+    ], 'Save');
+    $uuid = $this->getUuid();
+    $this->assertNotEmpty($uuid);
+    $this->cleanUp[] = $uuid;
+
+    /** @var \Drupal\Core\Extension\ModuleInstallerInterface $installer */
+    $installer = \Drupal::service('module_installer');
+    $installer->install(['devportal_api_reference']);
```

Comment:
Router may needs to be rebuild after you enabled a module.
https://github.com/apigee/apigee-edge-drupal/blob/fe6f716c193c5487fa162ee285dcb3ccee4d5085/tests/src/Functional/ApigeeEdgeTestTrait.php#L319